agent/src/flow_generator/protocol_logs/mq/kafka.rs

```diff
@@ -555,17 +681,64 @@ impl KafkaLog {
         &mut self,
         api_key: u16,
         api_version: u16,
-        code: i16,
+        payload: &[u8],
         info: &mut KafkaInfo,
+        code: Option<i16>,
     ) {
-        if api_key == KAFKA_FETCH && api_version >= 7 {
-            info.status_code = Some(code as i32);
+        if let Some(code) = code {
             if code == 0 {
                 info.status = L7ResponseStatus::Ok;
+                info.status_code = Some(code);
             } else {
                 info.status = L7ResponseStatus::ServerError;
                 self.perf_stats.as_mut().map(|p| p.inc_resp_err());
             }
+            return;
+        }
+        let Some(mut fixed_offset) = kafka_apiversion_errcode_fixed_offset(api_key, api_version)
+        else {
+            return;
+        };
+        if fixed_offset > payload.len() {
+            return;
+        }
+        let mut topic_len = 0;
+        match api_key {
+            KAFKA_PRODUCE => {
+                if api_version <= 8 {
+                    topic_len = read_i16_be(&payload[4..6]);
+                }
+                //Significant improvements since version 9, does not support now
+                if api_version == 9 {
+                    return;
+                };
+            }
+            KAFKA_FETCH => {
+                if api_version == 0 {
+                    topic_len = read_i16_be(&payload[4..6]);
+                } else if api_version <= 6 {
+                    topic_len = read_i16_be(&payload[8..10]);
+                } else if api_version >= 12 {
+                    // The 12th version is a transitional version, and the decoding protocols of the previous versions are more different.
+                    return;
+                }
+            }
+            _ => {
+                return;
+            }
+        }
```

Comment:
```suggestion
        let topic_len = match api_key {
            KAFKA_PRODUCE => {
                if api_version <= 8 {
                    read_i16_be(&payload[4..6])
                } else if api_version == 9 {
                    //Significant improvements since version 9, does not support now
                    return;
                }
            }
            KAFKA_FETCH => {
                if api_version == 0 {
                    read_i16_be(&payload[4..6])
                } else if api_version <= 6 {
                     read_i16_be(&payload[8..10])
                } else if api_version >= 12 {
                    // The 12th version is a transitional version, and the decoding protocols of the previous versions are more different.
                    return;
                }
            }
            _ => {
                return;
            }
        }
```